src/KafkaW/Producer.h

```diff
@@ -1,77 +1,44 @@
 #pragma once
 
 #include "BrokerSettings.h"
-#include "ConsumerMessage.h"
+#include "KafkaEventCb.h"
+#include "ProducerDeliveryCb.h"
+#include "ProducerMessage.h"
+#include "ProducerStats.h"
 #include <atomic>
 #include <functional>
-#include <librdkafka/rdkafka.h>
 
 namespace KafkaW {
 
 class ProducerTopic;
 
-class ProducerMsg {
-public:
-  virtual ~ProducerMsg() = default;
-  uchar *data;
-  uint32_t size;
-};
-
-struct ProducerStats {
-  std::atomic<uint64_t> produced{0};
-  std::atomic<uint32_t> produce_fail{0};
-  std::atomic<uint32_t> local_queue_full{0};
-  std::atomic<uint64_t> produce_cb{0};
-  std::atomic<uint64_t> produce_cb_fail{0};
-  std::atomic<uint64_t> poll_served{0};
-  std::atomic<uint64_t> msg_too_large{0};
-  std::atomic<uint64_t> produced_bytes{0};
-  std::atomic<uint32_t> out_queue{0};
-  ProducerStats() = default;
-  ProducerStats(ProducerStats const &) = default;
-};
-
 class ProducerInterface {
 public:
   ProducerInterface() = default;
   virtual ~ProducerInterface() = default;
   virtual void poll() = 0;
-  virtual uint64_t outputQueueLength() = 0;
+  virtual int outputQueueLength() = 0;
+  virtual RdKafka::Producer *getRdKafkaPtr() const = 0;
+  ProducerStats Stats;
 };
 
 class Producer : public ProducerInterface {
 public:
   typedef ProducerTopic Topic;
```

Comment:
Do not use "typedef".
```suggestion
    using Topic = ProducerTopic;
```